docs/DeveloperGuide.md

```diff
@@ -250,44 +263,66 @@ _{Explain here how the data archiving feature will be implemented}_
+| `* * *`  | student                                       | filter tutors by their subject discipline                                                     | find a tutor that caters to my needs (academic)                                           |
+| `* * *`  | student                                       | filter tutor by cost                                                                          | find a tutor that fits into budget as well                                                |
+| `* * *`  | student                                       | filter a tutor by his/her name                                                                | view his/her profile                                                                      |
+| `* * *`  | student                                       | filter tutors by their years of experience                                                    | find a tutor with experience within the range of my expectation                           |
+| `* * *`  | student                                       | filter tutors by their available timeslots                                                    | find a tutor with matched tuition time                                                    |
+| `* * *`  | student                                       | filter tutors by their available location                                                     | find tutors in a specific area                                                            |
 *{More to be added}*
 
 ### Use cases
 
 (For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: Delete a person**
-
-**MSS**
-
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
-
+### Appointment
+<pre>
+Use Case UC001: Add an appointment
+</pre>
+<pre>
+Use Case UC002: Delete an appointment
+
+<b>MSS</b>
+1. User requests to list persons
+2. AddressBook shows a list of persons
+3. User requests to delete a specific person in the list
+4. AddressBook deletes the person.
+Use case ends.
+
+<b>Extensions</b>
+2a. The list is empty.
     Use case ends.
 
-**Extensions**
+3a. The given index is invalid.
+    3a1. AddressBook shows an error message.
+    Use case resumes at step 2.
 
-* 2a. The list is empty.
+<b>{More to be added}</b>
+</pre>
 
-  Use case ends.
+<pre>
+Use Case UC003: List tuition appointments
+View the list of personal tuition appointments.
 
-* 3a. The given index is invalid.
+<b>System: Tutor Tracker</b>
+<b>Actor: User</b>
 
-    * 3a1. AddressBook shows an error message.
+<b>Preconditions: Appointment list must exist.</b>
```

Comment:
Is this precondition required? I think it is fair to assume that a list always exists and it might just be empty.

Reply:
Noted, I have removed it & commited.